src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
+        int oldCapacity = array.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
+        }
+        if (oldCapacity > 0 || array != empty_arraydata) {
+            int newCapacity = (int) (oldCapacity * multiplicator + 1);
+            return array = Arrays.copyOf(array, newCapacity);
+        } else {
+            return array = (T[]) new Object[def_capacity];
+        }
+    }
+
+
+    public static Object[] listToArray(List<?> list) {
+        if (!list.isEmpty()) {
+            Object[] tempArray = new Object[list.size()];
+            for (int index = 0; index < list.size(); index++) {
+                tempArray[index] = list.get(index);
+            }
+            return tempArray;
+        }
+        return null;
```

Comment:
**Suggestion** Returning null values from methods is generally considered bad practice and should be avoided if possible. You could return an empty array instead of null, which will give more predictable outcome for the users of this public API and will not force them to use extremely defensive programmings style, where all the time they have to check if the value returned is null and handle it

Reply:
Potential remedy to avoiding null here:

```suggestion
    Object[] tempArray = new Object[list.size()];
    for (int index = 0; index < list.size(); index++) {
        tempArray[index] = list.get(index);
    }
    return tempArray;
```